### ISSUE TITLE
Design of network regarding history and future GT

### ISSUE THREAD

**Issue by VVeiCao** [jchengai/forecast-mae#20]
Hi,

Thank you for the nice work!
I have a question regarding the design of history and future GT:
1. As i can see from the code: the history feature covers position/velocity displacement and mask, but the future feature only covers the position displacement and mask? What is the intuition?
```
         hist_feat = torch.cat(
            [
                data["x"], # displacement
                data["x_velocity_diff"][..., None],
                ~hist_padding_mask[..., None],
            ],
            dim=-1,
        ) # torch.Size([92, 50, 4])
        B, N, L, D = hist_feat.shape
        hist_feat = hist_feat.view(B * N, L, D)
        hist_feat = self.hist_embed(hist_feat.permute(0, 2, 1).contiguous())
        hist_feat = hist_feat.view(B, N, hist_feat.shape[-1]) # torch.Size([2, 46, 128])

        future_padding_mask = data["x_padding_mask"][:, :, 50:]
        future_feat = torch.cat([data["y"], ~future_padding_mask[..., None]], dim=-1)
```
3.  the GT of the history is `x = (data["x_positions"] - data["x_centers"].unsqueeze(-2)).view(-1, 50, 2)` which is the relative positions regarding the position of ego in current(T=50) frame, but the GT for the future is the position displacement. Also what is the intuition behind that? Why the history and future trajectory should have different design? 
```
 # hist pred loss
        x_hat = self.history_pred(hist_token).view(-1, 50, 2)
        x = (data["x_positions"] - data["x_centers"].unsqueeze(-2)).view(-1, 50, 2)
        x_reg_mask = ~data["x_padding_mask"][:, :, :50]
        x_reg_mask[~hist_pred_mask] = False
        x_reg_mask = x_reg_mask.view(-1, 50)
        hist_loss = F.l1_loss(x_hat[x_reg_mask], x[x_reg_mask])

        # future pred loss
        y_hat = self.future_pred(future_token).view(-1, 60, 2)  # B*N, 120
        y = data["y"].view(-1, 60, 2)
        reg_mask = ~data["x_padding_mask"][:, :, 50:]
        reg_mask[~future_pred_mask] = False
        reg_mask = reg_mask.view(-1, 60)
        future_loss = F.l1_loss(y_hat[reg_mask], y[reg_mask])
```
Thanks!

**Comment by skylineo** [comment]
Yeah，I have the same question as well. Have you found the answer to the issue? BTW, I have another question: Is it reasonable to use the same number of epochs and learning rate for both pre-training and fine-tuning in the article? Should the learning rate and number of epochs for pre-training be slightly higher to prevent excessive modification of pre-trained weights during fine-tuning? Looking forward to your reply!
